frontend/next.config.js

```diff
@@ -106,6 +106,27 @@ module.exports = withBundleAnalyzer({
       type: "asset/source",
     });
 
+    // msw/browser is not exported when compiling for Node (see
+    // https://github.com/mswjs/msw/blob/461a1885280451ec0837fdce897f49521b8ff260/package.json#L24)
+    // and msw/node is not exported when compiling for the browser (see
+    // https://github.com/mswjs/msw/blob/461a1885280451ec0837fdce897f49521b8ff260/package.json#L17),
+    // so to avoid build errors, we have to prevent Webpack from trying to
+    // traverse them when building.
+    // See https://github.com/mswjs/msw/issues/1801#issuecomment-1793911389
+    if (options.isServer) {
+      if (Array.isArray(config.resolve.alias)) {
+        config.resolve.alias.push({ name: "msw/browser", alias: false });
+      } else {
+        config.resolve.alias["msw/browser"] = false;
+      }
+    } else {
+      if (Array.isArray(config.resolve.alias)) {
+        config.resolve.alias.push({ name: "msw/node", alias: false });
+      } else {
+        config.resolve.alias["msw/node"] = false;
+      }
+    }
+
     return config;
```

Comment:
suggestion (totally-optional non-blocking): I treat every PR with an `else` statement like a little code-kata for me to practice refactoring into an early-return function instead.

```suggestion
    if (options.isServer) {
      if (Array.isArray(config.resolve.alias)) {
        config.resolve.alias.push({ name: "msw/browser", alias: false });
        return config;
      }
      config.resolve.alias["msw/browser"] = false;
      return config;
    }
    if (Array.isArray(config.resolve.alias)) {
      config.resolve.alias.push({ name: "msw/node", alias: false });
      return config;
    }
    config.resolve.alias["msw/node"] = false;
    return config;
```

Reply:
I usually lean towards early-return as well, but in this case, modifying `config.resolve.alias` is not the only purpose of this function, and returning early could hamper future additions to the config.

It could work if I extracted this into a separate function, but... That seems like the indirection might harm readability more?